libraries/protocol/include/steem/protocol/asset_symbol.hpp

```diff
@@ -207,20 +216,34 @@ inline void to_variant( const steem::protocol::asset_symbol_type& sym, fc::varia
 {
    try
    {
-      std::vector< variant > v( 2 );
-      v[0] = sym.decimals();
-      v[1] = sym.to_nai_string();
+      mutable_variant_object o;
+         o( ASSET_SYMBOL_NAI_KEY, sym.to_nai_string() )
+          ( ASSET_SYMBOL_DECIMALS_KEY, sym.decimals() );
+      var = std::move( o );
    } FC_CAPTURE_AND_RETHROW()
 }
 
 inline void from_variant( const fc::variant& var, steem::protocol::asset_symbol_type& sym )
 {
+   using steem::protocol::asset_symbol_type;
+
    try
    {
-      auto v = var.as< std::vector< variant > >();
-      FC_ASSERT( v.size() == 2, "Expected tuple of length 2." );
+      FC_ASSERT( var.is_object(), "Expected an object." );
+
+      auto& o = var.get_object();
+
+      auto nai = o.find( ASSET_SYMBOL_NAI_KEY );
+      FC_ASSERT( nai != o.end(), "Expected key '${key}'.", ("key", ASSET_SYMBOL_NAI_KEY) );
+      FC_ASSERT( nai->value().is_string(), "Expected a string type." );
```

Comment:
Say which field you expected a string type in. (i.e. "Expected a string type for value 'nai'"). Do the same thing for the decimals field. Also, please check the asset serialization and if such error messages don't exist there, please add them while you're in this code.